docs/public/specifications/test-zones/DNSSEC-TP/dnssec01.md

```diff
@@ -0,0 +1,311 @@
+  * The algorithm of the DS digest in the response is 2.
+
+### ALGO-OK-4
+The DS digest algorithm number is 4
+
+* Zone: algo-ok-4.dnssec01.xa.
+  * The algorithm of the DS digest in the response is 4.
+
+### ALGO-OK-5
+The DS digest algorithm number is 5
+
+* Zone: algo-ok-5.dnssec01.xa.
+  * The algorithm of the DS digest in the response is 5.
+
+### ALGO-OK-6
+The DS digest algorithm number is 6
+
+* Zone: algo-ok-6.dnssec01.xa.
+  * The algorithm of the DS digest in the response is 6.
+
+### MIXED-ALGO-1
+Three DS with different algorithms.
+
+* Zone: mixed-algo-1.dnssec01.xa.
+  * The response has three DS with different algorithms:
+    * 1
+    * 2
+    * 253
+
+### SHARED-IP-1
+Two parent name servers resolv to the same IP address.
+
+* Zone: child.shared-ip-1.dnssec01.xa.
+  * The parent zone is "shared-ip-1.dnssec01.xa" and the grandparent zone is
+    "dnssec01.xa".
+  * The parent is delegated to IB ns1a and ns1b with the same IPv4 and IPv6
+    addresses.
+
+### SHARED-IP-2
+Delegation of parent does not use the same name server name as parent zone.
+
+* Zone: child.shared-ip-2.dnssec01.xa.
+  * The parent zone is "shared-ip-2.dnssec01.xa" and the grandparent zone is
+    "dnssec01.xa".
+  * The parent is delegated to IB ns1 and ns2, but in the parent zone the names
+    are dns1 and dns2.
+    * ns1 and dns1 resolve to the same IP addresses.
+    * ns2 and dns2 resolve to the same IP addresses.
+
+### NO-RESPONSE-1
+No response from any of the servers on the DS query.
+
+* Zone: child.no-response-1.dnssec01.xa.
+  * No response at all from parent ns1 and ns2.
+
+### NO-VALID-RESPONSE-1
+No valid response from any of the servers on the DS query.
+
+* Zone: child.no-valid-response-1.dnssec01.xa.
+  * Response from parent ns1 has RCODE SERVFAIL.
+  * Response from parent ns2 has RCODE REFUSED.
+  * Child zone does not exist
+
+### PARENT-SERVER-NO-DS-1
+No DS from parent ns1.
+
+* Zone: child.parent-server-no-ds-1.dnssec01.xa.
+  * Response from parent ns1 is NODATA (no DS).
+  * Response from parent ns2 is normal.
+
+### PARENT-ZONE-NO-DS-1
+No DS from neither parent ns1 nor parent ns2.
+
+* Zone: parent-zone-no-ds-1.dnssec01.xa.
+  * Responses from parent ns1 and parent ns2 are NODATA (no DS).
+
+### UNDEL-NO-UNDEL-DS-1
+Zone is not delegated, but undelegated data is provided. No DS.
+
+* Zone: undel-no-undel-ds-1.dnssec01.xa.
+  * The zone is not delegated, but there is undelegated data.
+  * ns1 and ns2 are OOB.
+  * No undelegated DS is provided.
+
+### UNDEL-WITH-UNDEL-DS-1
+Zone is not delegated, but undelegated data is provided with DS.
+
+* Zone: undel-with-undel-ds-1.dnssec01.xa.
+  * The zone is not delegated, but there undelegated data.
+  * ns1 and ns2 are OOB.
+  * Undelegated DS is provided.
+
+### ROOT-NO-UNDEL-DS-1
+Zone is the root zone, and no undelegated DS is provided.
+
+* Zone: "."
+  * ns1 and ns2 are IB
+  * The zone exists.
+
+### ROOT-WITH-UNDEL-DS-1
+Zone is the root zone, and undelegated DS is provided.
```

Comment:
What's the provided undelegated DS data supposed to be?

Reply:
It is provided in test-zones-output.md, not here.